api/oc_push.c

```diff
@@ -0,0 +1,2625 @@
+                      oc_string_len(rep->value.string));
+      }
+      break;
+
+    case OC_REP_STRING_ARRAY:
+      if (!strcmp(oc_string(rep->name), "rts")) {
+        oc_free_string_array(&recv_obj->rts);
+        int len = oc_string_array_get_allocated_size(rep->value.array);
+        oc_new_string_array(&recv_obj->rts, len);
+
+        for (int i = 0; i < len; i++) {
+          oc_string_array_add_item(
+            recv_obj->rts, oc_string_array_get_item(rep->value.array, i));
+        }
+      }
+      break;
+
+    default:
+      OC_PUSH_ERR("something wrong, unexpected Property type: %d", rep->type);
+      return;
+    }
+    rep = rep->next;
+  }
+
+  return;
+}
+
+/*
+ * @brief create & add new receiver object
+ *
+ * @param recvs_instance Resource representation of Push Receiver Resource
+ * @param rep received new receiver object object
+ * @return true:success, false:fail
+ */
+static bool
+_create_recv_obj(oc_recvs_t *recvs_instance, oc_rep_t *rep)
+{
+  bool result = false;
+  char mandatory_property_check = 0;
+  oc_recv_t *recv_obj = (oc_recv_t *)oc_memb_alloc(&recv_instance_memb);
+
+  if (!recv_obj) {
+    OC_PUSH_ERR("oc_memb_alloc() error!");
+    return result;
+  }
+
+  while (rep) {
+    switch (rep->type) {
+    case OC_REP_STRING:
+      if (!strcmp(oc_string(rep->name), "receiveruri")) {
+        oc_new_string(&recv_obj->receiveruri, oc_string(rep->value.string),
+                      oc_string_len(rep->value.string));
+        mandatory_property_check |= 0x1;
+      }
+      break;
+
+    case OC_REP_STRING_ARRAY:
+      if (!strcmp(oc_string(rep->name), "rts")) {
+        int len = oc_string_array_get_allocated_size(rep->value.array);
+        oc_new_string_array(&recv_obj->rts, len);
+
+        for (int i = 0; i < len; i++) {
+          oc_string_array_add_item(
+            recv_obj->rts, oc_string_array_get_item(rep->value.array, i));
+        }
+
+        mandatory_property_check |= 0x2;
+      }
+      break;
+
+    default:
+      OC_PUSH_ERR("something wrong, unexpected Property type: %d", rep->type);
+      break;
+    }
+    rep = rep->next;
+  }
+
+  if (mandatory_property_check != 0x3) {
+    oc_memb_free(&recv_instance_memb, recv_obj);
+    return result;
+  }
+
+  oc_list_add(recvs_instance->receivers, recv_obj);
+
+  /* create app resource corresponding to receiver object */
+  OC_PUSH_DBG("new app resource for new receiver obj (\"%s\") is created...",
+              oc_string(recv_obj->receiveruri));
+  if (_create_pushd_rsc(recv_obj, recvs_instance->resource))
+    result = true;
+
+  return result;
+}
+
+/*
+ * @brief validate if receiver object array has any problem or not
+ *
+ * @param obj_list receiver object array
+ * @return true:success, false:fail
+ */
+static bool
+_validate_recv_obj_list(oc_rep_t *obj_list)
+{
+  oc_rep_t *recv_obj, *rep;
+  bool result = false;
+  char mandatory_property_check;
+
+  if (!obj_list)
+    goto exit;
```

Comment:
(This is OK, though usually `goto` is used when you need to execute some calls before returning, here just pure `return false;` would be simpler)

Reply:
agreed, fixed.